docs/sources/administration/api-keys/index.md

```diff
@@ -44,28 +48,91 @@ This topic shows you how to create an API key using the Grafana UI. You can also
 
-- Ensure you have permission to create Grafana service accounts. For more information about permissions, refer to [Roles and permissions]({{< relref "../roles-and-permissions/#" >}}).
+#### Before you begin
 
-**To migrate all API keys to service accounts:**
+To follow these instructions, you need:
+
+- To ensure you have permission to create Grafana service accounts. For more information about permissions, refer to [Roles and permissions]({{< relref "../roles-and-permissions/#" >}}).
+
+#### Steps
+
+To migrate all API keys to service accounts, complete the following steps:
 
 1. Sign in to Grafana, hover your cursor over **Configuration** (the gear icon), and click **API Keys**.
-2. In the top of the page, find the section which says **Switch from API keys to service accounts**
-3. Click **Migrate to service accounts now**.
-4. A confirmation window will appear, asking to confirm the migration. Click **Yes, migrate now** if you are willing to continue.
-5. Once migration is successful, you can choose to forever hide the API keys page. Click **Hide API keys page forever** if you want to do that.
+1. In the top of the page, find the section which says **Switch from API keys to service accounts**
+1. Click **Migrate to service accounts now**.
+1. A confirmation window will appear, asking to confirm the migration. Click **Yes, migrate now** if you are willing to continue.
+1. Once migration is successful, you can choose to forever hide the API keys page. Click **Hide API keys page forever** if you want to do that.
 
-**To migrate single API key to a service account:**
+To migrate a single API key to a service account, complete the following steps:
 
 1. Sign in to Grafana, hover your cursor over **Configuration** (the gear icon), and click **API Keys**.
 1. Find the API Key you want to migrate.
 1. Click **Migrate to service account**.
+
+### Migrate API keys to Grafana service accounts using the API
+
+This section shows you how to migrate API keys to Grafana service accounts using the Grafana API.
+
+#### Before you begin
+
+To follow these instructions, you need:
+
+- xxx
+- xxx
+
+#### Steps
+
+Complete the following steps to migrate from API keys to service accounts using the API:
+
+1. Call the `POST /api/serviceaccounts` endpoint and the `POST /api/serviceaccounts/<id>/tokens`.
+
+   This action generates a service account token.
+
+1. Store the ID and secret that the system returns to you.
+1. Pass the token in the `Authrorization` header, prefixed with `Bearer`.
+
+   This action authenticates API requests.
+
+1. SATs used for authentication
+1. Remove code that handles the old `/api/auth/keys` endpoint.
+1. Track the [API keys](http://localhost:3000/org/apikeys) in use and migrate them to SATs.
+
+### Migrate API keys to Grafana service accounts using Terraform
+
+This section shows you how to migrate API keys to Grafana service accounts using Terraform.
+
+#### Before you begin
```

Comment:
@eleijonmarck If there are any prerequisites or special permissions a user must have in place, we can add them to this Before you begin section. Are there any such prereqs? If not, we can remove this section.

Reply:
from the title of this section, it is self evident that he user needs and probably have the terraform configuration is looking for how to migrate from that. So i would say the prerequisites is not needed here.

Reply:
Gotcha. Removing.